pkg/openwsn/contrib/radio_hal.c

```diff
@@ -0,0 +1,316 @@
+        break;
+    case IEEE802154_RADIO_INDICATION_TX_START:
+        openwsn_radio.startFrame_cb(_txrx_event_capture_time);
+        break;
+    case IEEE802154_RADIO_INDICATION_RX_START:
+        openwsn_radio.startFrame_cb(_txrx_event_capture_time);
+        break;
+    default:
+        break;
+    }
+}
+
+int openwsn_radio_init(void *radio_dev)
+{
+    assert(radio_dev);
+    ieee802154_dev_t *dev = (ieee802154_dev_t *)radio_dev;
+
+    LOG_DEBUG("[openwsn/radio]: initialize riot-adaptation\n");
+    openwsn_radio.dev = dev;
+
+    /* override short_address and panid*/
+    _idmanager_addr_override();
+
+    if (ieee802154_radio_request_on(dev)) {
+        LOG_ERROR("[openwsn/radio]: unable to initialize device\n");
+        return -1;
+    }
+
+    /* Set the Event Notification */
+    dev->cb = _hal_radio_cb;
+
+    /* If the radio is still not in TRX_OFF state, spin */
+    while (ieee802154_radio_confirm_on(dev) == -EAGAIN) {}
+
+    /* Enable basic mode, no AUTOACK. no CSMA */
+    ieee802154_radio_set_rx_mode(dev, IEEE802154_RX_AACK_DISABLED);
+    ieee802154_radio_set_csma_params(dev, NULL, -1);
+
+    /* MAC layer will handle retransmissions */
+    if (ieee802154_radio_has_frame_retrans(dev)) {
+        ieee802154_radio_set_frame_retrans(dev, 0);
+    }
+#ifdef MODULE_CC2538_RF
+    /* If frame filtering is enabled cc2538 will not accept beacons
+       where the destination-address mode is 0 (no destination address).
+       per rfc8180 4.5.1 the destination address must be set, which means
+       the destination-address mode can't be 0 */
+    ieee802154_radio_set_rx_mode(dev, IEEE802154_RX_PROMISC);
+#endif
+
+    /* Configure PHY settings (channel, TX power) */
+    ieee802154_phy_conf_t conf =
+    { .channel = CONFIG_IEEE802154_DEFAULT_CHANNEL,
+      .page = CONFIG_IEEE802154_DEFAULT_CHANNEL,
+      .pow = CONFIG_IEEE802154_DEFAULT_TXPOWER };
+
+    ieee802154_radio_config_phy(dev, &conf);
+
+    return 0;
+}
+
+void radio_setStartFrameCb(radio_capture_cbt cb)
+{
+    openwsn_radio.startFrame_cb = cb;
+}
+
+void radio_setEndFrameCb(radio_capture_cbt cb)
+{
+    openwsn_radio.endFrame_cb = cb;
+}
+
+void radio_reset(void)
+{
+    /* TODO: this is not handled correctly since not all radios implement
+       this */
+    ieee802154_radio_off(openwsn_radio.dev);
+    ieee802154_radio_request_on(openwsn_radio.dev);
+    /* If the radio is still not in TRX_OFF state, spin */
+    while (ieee802154_radio_confirm_on(openwsn_radio.dev) == -EAGAIN) {}
+}
+
+void radio_setFrequency(uint8_t frequency, radio_freq_t tx_or_rx)
+{
+    (void)tx_or_rx;
+
+    ieee802154_phy_conf_t conf =
+    { .channel = frequency,
+      .page = CONFIG_IEEE802154_DEFAULT_CHANNEL,
+      .pow = CONFIG_IEEE802154_DEFAULT_TXPOWER };
+
+    ieee802154_radio_config_phy(openwsn_radio.dev, &conf);
+}
+
+void radio_rfOn(void)
+{
+    ieee802154_radio_request_on(openwsn_radio.dev);
+    ieee802154_radio_request_set_trx_state(openwsn_radio.dev,
```

Comment:
Two things here:
1) The `ieee802154_radio_request_set_trx_state` function doesn't have its counterpart `ieee802154_radio_confirm_set_trx_state`. This might produce an inconsistent state in some Radio HAL implementations
2) Anyway, the HAL API guarantees that the is in `TRX_OFF` after `ieee802154_radio_confirm_on`. So, you can remove this line anyway

Reply:
I think I had it because the `cc2538` does not have a proper on function (and I was lazy to implement it (or better said would rather do it another PR))

Reply:
I'll add a comment for the momment if you don't mind

Reply:
did just that